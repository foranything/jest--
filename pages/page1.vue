<template>
  <div id="page1-id" class="bar">
    <MessageList :messages="messages"/>
  </div>
</template>

<script>
import MessageList from '@/components/MessageList'

export default {
  name: 'app',
  data: () => ({ messages: ['Hey John', 'Howdy Paco'] }),
  components: {
    MessageList
  }
}
</script>