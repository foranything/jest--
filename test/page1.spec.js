import Vue from "vue";
import page1 from "@/pages/page1.vue";

describe('page1', () => {
    let cmp, vm;

    beforeEach(() => {
      cmp = Vue.extend(page1); // Create a copy of the original component
      vm = new cmp({
        data: {
          // Replace data value with this fake data
          messages: ["Cat", "Dog"]
        }
      }).$mount(); // Instances and mounts the component
    });

    it('equals messages to ["Cat", "Dog"]', () => {
        expect(vm.messages).toEqual(["Cat", "Dog"]);
    });

    it("has the expected html structure", () => {
        expect(vm.$el).toMatchSnapshot();
    });
    // snapshot을 파일로 보관
    // snapshot이 없을 경우: 새로운 스냅샷 생성, error 없음
    // snapshot이 이미 있고, 스냅샷이 같은 경우: test 통과, error 없음
    // snapshot이 이미 있고, 스냅샷이 다른 경우: text 실패, error, yarn test -u 명령어로 스냅샷 업데이트 가능
    // 문제점: page1에 대한 독립적인 테스트이기 때문에 자식 컴포넌트에서 문제가 생길 시 원하지 않는 결과가 나올 수 있음
})
