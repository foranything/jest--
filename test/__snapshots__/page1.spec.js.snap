// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page1 has the expected html structure 1`] = `
<div
  id="app"
>
  <ul>
    <li>
      
    Cat
  
    </li>
    <li>
      
    Dog
  
    </li>
  </ul>
</div>
`;
